etc/kayobe/environments/ci-aio/automated-setup.sh

```diff
@@ -1,109 +1,333 @@
+    if type dnf > /dev/null 2>&1; then
+        sudo dnf -y install git python3.12
+    else
+        sudo apt update
+        sudo apt -y install gcc git libffi-dev python3.12-dev python-is-python3 python3.12-venv
+    fi
+}
+
+# Clone source repositories
+clone_repositories() {
+    echo "Cloning repositories..."
+    mkdir -p "$BASE_PATH/src"
+    pushd "$BASE_PATH/src"
+    # NOTE: Kayobe isn't used for deploying the AIO, but it is used for
+    # running the overcloud test VM script. TODO: run installed version
+    [[ -d kayobe ]] || git clone https://github.com/stackhpc/kayobe.git -b "$KAYOBE_BRANCH"
+    [[ -d kayobe-config ]] || git clone https://github.com/stackhpc/stackhpc-kayobe-config kayobe-config -b "$KAYOBE_CONFIG_BRANCH"
+    popd
+}
+
+# Pause for configuration editing if requested
+pause_for_configuration() {
+    if [[ "$KAYOBE_CONFIG_EDIT_PAUSE" == true ]]; then
+        echo "Deployment is paused, edit configuration in another terminal"
+        echo "Press enter to continue"
+        read -rs
+    fi
+}
+
+# Disable OVN if requested (aka deploy OVS)
+disable_ovn() {
+    if [[ "$USE_OVS" == true ]]; then
+        echo "Disabling OVN..."
+        echo "kolla_enable_ovn: false" > "$BASE_PATH/src/kayobe-config/etc/kayobe/environments/ci-aio/disable-ovn.yml"
+    fi
+}
+
+# Remove LVM configuration if not using LVM
+remove_lvm_configuration() {
+    if ! sudo vgdisplay | grep -q lvm2; then
+        echo "Removing LVM configuration..."
+        rm -f "$BASE_PATH/src/kayobe-config/etc/kayobe/environments/ci-aio/inventory/group_vars/controllers/lvm.yml" || true
+        sed -i -e '/controller_lvm_groups/,+2d' "$BASE_PATH/src/kayobe-config/etc/kayobe/environments/ci-aio/controllers.yml" || true
+    fi
+}
+
+# Set up Python virtual environment for Kayobe
+setup_virtualenv() {
+    echo "Setting up Python virtual environment..."
+    mkdir -p "$BASE_PATH/venvs"
+    pushd "$BASE_PATH/venvs"
+    if [[ ! -d kayobe ]]; then
+        python3.12 -m venv kayobe
+    fi
+    source kayobe/bin/activate
+    pip install -U pip
+    pip install -r "$BASE_PATH/src/kayobe-config/requirements.txt"
+    deactivate
+    popd
+}
+
+# AIO networking setup
+configure_network() {
+    echo "Configuring network..."
+    if ! ip l show breth1 >/dev/null 2>&1; then
+        sudo ip l add breth1 type bridge
+    fi
+    sudo ip l set breth1 up
+    if ! ip a show breth1 | grep 192.168.33.3/24; then
+        sudo ip a add 192.168.33.3/24 dev breth1
+    fi
+    if ! ip l show dummy1 >/dev/null 2>&1; then
+        sudo ip l add dummy1 type dummy
+    fi
+    sudo ip l set dummy1 up
+    sudo ip l set dummy1 master breth1
+}
+
+# Helper function to activate the Kayobe environment
+activate_kayobe_env() {
+    KAYOBE_VAULT_PASSWORD=$(cat $VAULT_PASSWORD_FILE)
+    export KAYOBE_VAULT_PASSWORD
+    pushd "$BASE_PATH/src/kayobe-config"
+    source kayobe-env --environment ci-aio
+    source $BASE_PATH/venvs/kayobe/bin/activate
```

Comment:
Could be good to add kayobe autocompletion here too, like we do in beokay env-vars. Just to test that activating autocompletion works.
```suggestion
    source $BASE_PATH/venvs/kayobe/bin/activate
    source <(kayobe complete)
```

Could be even better to test using beokay in this script, but that's scope creep for this PR.